File: benchmark/results/java-invocation-inline/run-004/b-baseline/output/com/metaengine/demo/identity/services/UserRepository.java
package com.metaengine.demo.identity.services;

import java.util.List;
import com.metaengine.demo.identity.aggregates.User;

/** UserRepository persistence service. */
public class UserRepository {
    /** Persist a new User. */
    public User create(User input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find a User by id; returns null if not found. */
    public User findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List up to limit Users. */
    public List<User> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a User by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-multilang/run-003/b-baseline/output/metaengine_demo/identity/services/user_repository.py
"""UserRepository for the identity domain."""
from typing import Optional

from metaengine_demo.identity.aggregates.user import User


class UserRepository:
    """UserRepository persistence service."""

    def create(self, input: Optional[User]) -> User:
        """Persist a new user."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[User]:
        """Find a user by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[User]:
        """List users up to a limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a user by id."""
        raise NotImplementedError("not implemented")
